File: design-probes/default-run/flows/greet/flow.ts
#!/usr/bin/env bun

// DESIGN PROBE ONLY: complete pseudocode against the future portable SDK.
import { serveRun } from "@flowmd/sdk";

interface GreetInput {
  readonly name: string;
}

serveRun<GreetInput>(run => ({
  outcome: "done",
  output: {
    greeting: `Hello, ${run.input.name}!`,
  },
}));

File: design-probes/dynamic-defaults/flows/echo-reverse/flow.py
#!/usr/bin/env python3

# DESIGN PROBE ONLY: future Python projection of FLOW Run/1.
from flowmd_sdk import serve_run


async def reverse_text(run):
    return {
        "outcome": "done",
        "output": {"value": "".join(reversed(run.input["text"]))},
    }


serve_run(reverse_text)
